pkg/execution/batch/batch.go

```diff
@@ -31,10 +31,10 @@ import (
 //	#4 needs to happen in one transaction in order to make sure there will not be any race conditions.
 type BatchManager interface {
 	Append(ctx context.Context, bi BatchItem, fn inngest.Function) (*BatchAppendResult, error)
-	RetrieveItems(ctx context.Context, batchID ulid.ULID) ([]BatchItem, error)
-	StartExecution(ctx context.Context, batchID ulid.ULID, batchPointer string) (string, error)
+	RetrieveItems(ctx context.Context, functionId uuid.UUID, batchID ulid.ULID, batchPointer string) ([]BatchItem, error)
```

Comment:
We need to include batch pointers for most operations as they will tell us whether the batch, at the time of appending, was sharded or now.